### PR TITLE
Update flag names to --request and --limit

### PR DESCRIPTION
**Pull request by navidshaikh**
## Description
 - Use singular flag names and support comma separated or repeated flag values
 - Update tests and docs

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -12,6 +12,17 @@
...
+
+[cols="1,10,3", options="header", width="100%"]
+|===
+| | Description | PR
+
+| 🐛
+| Update flag names to --request and --limit
+| https://github.com/knative/client/pull/872[#872]
+|===
+
```
Comment (rhuss) [inline CHANGELOG.adoc:25]: I think we can remove the v0.15.0 header as this version has never been officially released.

File: docs/cmd/kn_service_create.md [hunk 2]
```diff
@@ -67,9 +67,9 @@ kn service create NAME --image IMAGE [flags]
...
       --label-revision stringArray    Revision label to set. name=value; you may provide this flag any number of times to set multiple labels. To unset, specify the label name followed by a "-" (e.g., name-). This flag takes precedence over "label" flag.
       --label-service stringArray     Service label to set. name=value; you may provide this flag any number of times to set multiple labels. To unset, specify the label name followed by a "-" (e.g., name-). This flag takes precedence over "label" flag.
-      --limits string                 The resource requirement limits for this Service. For example, 'cpu=100m,memory=256Mi'.
-      --limits-cpu string             DEPRECATED: please use --limits instead. The limits on the requested CPU (e.g., 1000m).
-      --limits-memory string          DEPRECATED: please use --limits instead. The limits on the requested memory (e.g., 1024Mi).
+      --limit strings                 The resource requirement limits for this Service. For example, 'cpu=100m,memory=256Mi'.
```
Comment (rhuss) [inline docs/cmd/kn_service_create.md]: why is the type here still `strings` and not `stringArray` as for `label` ?

**Comment by rhuss** [comment]
It's possible that labels use an array whereas limits/requests use a slice. 

I think we should later adapt labels etc. you slices, too. + we should think about how we can share code better to avoid that kind of differences in the future.

File: docs/cmd/kn_service_create.md (hunk 2 above)
Comment (navidshaikh) [reply docs/cmd/kn_service_create.md]: if we use `StringSliceVar` type flags, we get `strings` (see traffic flags) and
if we use `StringArrayVar` type flags, we get `stringArray`

**Comment by navidshaikh** [comment]
> It's possible that labels use an array whereas limits/requests use a slice.

Yes (see traffic flags)

> I think we should later adapt labels etc. you slices, too. + we should think about how we can share code better to avoid that kind of differences in the future.

Yes, we'd need to fix this for rest of the flags to work as mentioned in the examples of convention at https://github.com/knative/client/blob/master/conventions/cli.md#maps

File: CHANGELOG.adoc (hunk 1 above)
Comment (navidshaikh) [reply CHANGELOG.adoc]: yup, fixed.

File: pkg/kn/flags/resources_test.go [hunk 3]
```diff
@@ -37,49 +37,75 @@ func parseQuantity(value string) resource.Quantity {
...
+			corev1.ResourceList{corev1.ResourceName("cpu"): parseQuantity("200m")},
+			false,
+		},
 	}
 	for _, c := range cases {
 		options := &ResourceOptions{}
 		options.Requests = c.requests
 		options.Limits = c.limits
-		err := options.Validate()
+		reqToRemove, limToRemove, err := options.Validate()
```
Comment (rhuss) [inline pkg/kn/flags/resources_test.go:91]: Maybe don't call it validate as the main purpose is no to extract the values. What's about `options.Extract()` or `options.Parse()` (or `options.ParseAndValidate()`) ?

Comment (navidshaikh) [reply pkg/kn/flags/resources_test.go:91]: Sure, will update the name when we unify the other similar flags.

**Pull request merged**